Prometheus changes (#252)
minio/cookbook: docs/how-to-monitor-minio-with-prometheus.md

MinIO server release `RELEASE.2018-05-11T00-29-24Z` or later running. To get started with MinIO, refer [MinIO QuickStart Document](https://docs.min.io/docs/minio-quickstart-guide). Follow below steps to get started with MinIO monitoring using Prometheus.

### 1. Configure Prometheus in MinIO

Prometheus is enabled and the authentication is set to "public" by default in MinIO configuration. For a secured access, update the prometheus config as follows

``` json
    "prometheus": {
        "auth": "jwt",
        "enable": true
    }

> ```

Reviewer: Need more specific information on setting this via `mc admin config set`, or reference to config documents.

Author: Since we no more rely on the server config, we dont deal with config sets/gets

To disable prometheus, set `enable` to `false`.

> To generate a bearer jwt token for an alias, use [mc](https://docs.min.io/docs/minio-client-quickstart-guide) like `mc admin prometheus jwt <alias>`.

Reviewer: After setting jwt token, we give no information on how to access metrics with the token

Author: We talk about the dashboard in 4rth point here - `4. Start Prometheus`. This is more self-explanatory. I think we dont need to give them much details about the prometheus dashboard. That can be inconsistent with the dashboard updates. They can still explore/read about it in prometheus docs.

> To generate a bearer jwt token for an alias, use [mc](https://docs.min.io/docs/minio-client-quickstart-guide) like `mc admin prometheus jwt <alias>`.

Reviewer: ```suggestion
To generate a bearer jwt token for an alias, use [mc](https://docs.min.io/docs/minio-client-quickstart-guide) as follows: `mc admin prometheus jwt <alias>`.
```

Author: :+1:

Refer [Prometheus documentation](https://prometheus.io/docs/introduction/first_steps/) for more details.

### 2. Generate Prometheus Config with mc

> As Prometheus endpoint in MinIO is authorized. For a secured access using a bearer token approach, override the default prometheus config with the one generated using mc.

Reviewer: ```suggestion
The Prometheus endpoint in MinIO requires authorization. For secured access using a bearer token approach, override the default Prometheus config with the one generated using mc.
```

> To generate a prometheus config for an alias, use [mc](https://docs.min.io/docs/minio-client-quickstart-guide) as follows `mc admin prometheus generate <alias>`.

Reviewer: Capitalization is not consistent here, if referring to the product should be capitalized, vs when added as part of a command (in this example, `...generate a Prometheus config` vs `mc admin prometheus`.

> The command will generate the `scrape_configs` section of the prometheus.yml as follows

Reviewer: ```suggestion
The command will generate the `scrape_configs` section of the prometheus.yml as follows:
```

```yaml
scrape_configs:
- job_name: minio-job
  bearer_token: <secret>
  metrics_path: /minio/prometheus/metrics
  scheme: http
  static_configs:
  - targets: ['localhost:9000']
```

### 3. Update `scrape_configs` section in prometheus.yml

To authorize every scrape request, copy and paste the generated `scrape_configs` section in the prometheus.yml and restart the prometheus service.

### 4. Start Prometheus

Start (or) Restart Prometheus service by running

```sh
./prometheus --config.file=prometheus.yml
```
Here `prometheus.yml` is the name of configuration file. You can now see MinIO metrics in Prometheus dashboard. By default Prometheus dashboard is accessible at `http://localhost:9090`.

> ## List of MinIO metric exposed

Author: I will send a seperate PR for the latest list of metrics. After the refactor PR gets merged.
